### PR TITLE
stellar-etl: implement export_trades command

### LINKED ISSUE #58: [1] cmd: implement command for exporting trades
### What
We need to implement the planned functionality for exporting trades. The command will use the `transform` package functions for transforming data. Part of #53.

### Why
This is one step towards having a functional pipeline for data transfer. The command will provide users with a way to access regularly updated trade data.

### PR DESCRIPTION
**Pull request by Isaiah-Turner**
<details>
  <summary>PR Checklist</summary>
  
### PR Structure

* [x] This PR has reasonably narrow scope (if not, break it down into smaller PRs).
* [x] This PR avoids mixing refactoring changes with feature changes (split into two PRs
  otherwise).
* [x] This PR's title starts with name of package that is most changed in the PR, ex.
  `services/friendbot`, or `all` or `doc` if the changes are broad or impact many
  packages.

### Thoroughness

* [x] This PR adds tests for the most critical parts of the new functionality or fixes.
* [x] I've updated any docs ([developer docs](https://www.stellar.org/developers/reference/), `.md`
  files, etc... affected by this change). Take a look in the `docs` folder for a given service,
  like [this one](https://github.com/stellar/go/tree/master/services/horizon/internal/docs).
</details>

### What
This PR implements the export_trades command. It allows users to connect to the history archives, retrieve trades information in a desired range, and export the JSON encodings to an output file. Closes #58.

### Why
This command is part of the larger CLI tool. It will allow for updating the BigQuery table trades more efficiently.

### Known limitations
The start-time and end-time flags are not implemented. The ingestion system uses sequence numbers instead of times. If it is expanded to use times, then we can add the flags. The serialize-method flag also is not implemented, since at the moment we only have json encodings for the structs.

### CODE REVIEW

File: cmd/export_trades.go [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+		}
+	},
+}
+
+func init() {
+	rootCmd.AddCommand(tradesCmd)
+	utils.AddBasicFlags("trades", tradesCmd.Flags())
+	tradesCmd.MarkFlagRequired("end-ledger")
+
+	/*
```
Comment (debnil) [inline cmd/export_trades.go:62]: Either remove this comment; edit it to only include TODOs; or move it above the `tradesCmd` declaration, so it's easier for the user to understand the command's semantics.

File: internal/input/trades.go [hunk 2]
```diff
@@ -0,0 +1,97 @@
...
+			tx, err := txReader.Read()
+			if err == ingestio.EOF {
+				break
+			}
+
+			for index, op := range tx.Envelope.Operations() {
+				// Trades occur on these operation types: manage buy offer, manage sell offer,
+				// create passive sell offer, path payment send, and path payment receive
+				// Trades also can only occur when these operations are successful
+				if operationResultsInTrade(op) && tx.Result.Successful() {
```
Comment (debnil) [inline internal/input/trades.go:59]: These are necessary conditions, but are these sufficient for an operation to be a trade? For example, wouldn't this mean a successful offer operation would count as a trade?

File: cmd/export_trades.go [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+			cmdLogger.Fatal("could not read trades: ", err)
+		}
+
+		for _, tradeInput := range trades {
+			trades, err := transform.TransformTrade(tradeInput.OperationIndex, tradeInput.Transaction, tradeInput.CloseTime)
+			if err != nil {
+				cmdLogger.Fatal("could not transform trade ", err)
+			}
+
+			// Each operation, when transformed, results in multiple trades, each of which needs to be exported
```
[the remark this answers is not in the document]
Comment (Isaiah-Turner) [reply cmd/export_trades.go:38]: From how operation results are set up, each operation can have multiple trades. The parent transaction holds the results for each operation, but trades are only visible within the operation results.

File: internal/input/trades.go (hunk 2 above)
Comment (Isaiah-Turner) [reply internal/input/trades.go:59]: Not all of these operations will result in trades, but this is checked in TransformTrade (an empty slice is returned if no trades occurred) for a given op.

**Pull request merged**